psi4/driver/procrouting/proc.py

```diff
@@ -2691,6 +2695,35 @@ def run_bccd(name, **kwargs):
     optstash.restore()
     return ref_wfn
 
+def run_tdscf_energy(**kwargs):
+
+    # Get a wfn in case we aren't given one
+    ref_wfn = kwargs.get('ref_wfn', None)
+    if ref_wfn is None:
+        raise ValidationError("TDSCF: No reference wave function passed!")
+
+    states = core.get_global_option("TDSCF_STATES_PER_IRREP")
+
+    # some sanity checks
+    if sum(states) == 0:
+        raise ValidationError("TDSCF: No states requested in TDSCF_STATES_PER_IRREP") 
+
+    if len(states) != ref_wfn.nirrep():
+        raise ValidationError("TDSCF: Requested TDSCF_STATES_PER_IRREP do not match wave function symmetry") 
+
+    ssuper_name = ref_wfn.functional().name()
+
```

Comment:
We need a way to detect double-hybrid functionals. Possibly `SuperFunctional.c_alpha > 0`.
As a reminder, Grimme/Neese suggested to replace the MP2 part in the ground state with CIS(D) in the excited state.

We need to decide if we still calculate the TD-DFT part or exist with a warning right away. To avoid misunderstandings, I'd suggest to exist before the calculation starts with a warning that excited states with double-hybrids is not implemented.

Reply:
Exiting is the safest course of action, I think.

Reply:
https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/proc_table.py#L246 is a model for DH check